cpp/src/arrow/ipc/reader.cc

```diff
@@ -664,14 +690,15 @@ Result<std::shared_ptr<RecordBatch>> ReadRecordBatch(
   std::shared_ptr<Schema> out_schema;
   // Empty means do not use
   std::vector<bool> inclusion_mask;
-  RETURN_NOT_OK(GetInclusionMaskAndOutSchema(schema, options.included_fields,
+  DictionaryKind kind;
```

Comment:
kind is never used? See above comment re: this parameter

Reply:
Yes, it is not used, This is a dummy variable to allocate an IpcReadContext().